src/reducers/versions.spec.tsx

```diff
@@ -2470,4 +2524,206 @@ describe(__filename, () => {
       expect(getMostRelevantEntryStatus(version, parentDir)).toEqual('M');
     });
   });
+
+  describe('goToRelativeDiff', () => {
+    const _goToRelativeDiff = ({
+      _getRelativeDiff = jest.fn(),
+      _viewVersionFile = jest.fn(),
+      currentAnchor = '',
+      diff = null,
+      pathList = ['file1.js'],
+      position = RelativePathPosition.next,
+      versionId = 1,
+    } = {}) => {
+      return goToRelativeDiff({
+        _getRelativeDiff,
+        _viewVersionFile,
+        currentAnchor,
+        diff,
+        pathList,
+        position,
+        versionId,
+      });
+    };
+
+    it('dispatches push for a new anchor for the current file', async () => {
+      const anchor = 'D1';
+      const _getRelativeDiff = jest.fn().mockReturnValue({ anchor });
+      const currentAnchor = '';
+      const diff = null;
+      const pathList = ['file1.js'];
+      const position = RelativePathPosition.next;
+      const versionId = 123;
+
+      const store = configureStore();
+      store.dispatch(
+        actions.loadVersionInfo({ version: { ...fakeVersion, id: versionId } }),
+      );
+
+      const { dispatch, thunk } = thunkTester({
+        createThunk: () =>
+          _goToRelativeDiff({
+            _getRelativeDiff,
+            currentAnchor,
+            diff,
+            pathList,
+            position,
+            versionId,
+          }),
+        store,
+      });
+
+      const version = getVersionInfo(
+        store.getState().versions,
+        versionId,
+      ) as Version;
+
+      await thunk();
+
+      expect(_getRelativeDiff).toHaveBeenCalledWith({
+        currentAnchor,
+        diff,
+        pathList,
+        position,
+        version,
+      });
+
+      expect(dispatch).toHaveBeenCalledWith(
+        push(
+          // TODO: This is probably not good enough.
+          // We should check the location object as well, but I'm not sure how
+          // to set up a location in the router in the test.
```

Comment:
Did you try [`createFakeLocation()`](https://github.com/mozilla/addons-code-manager/blob/5ab911bda04346c6d5e9583f467992f9051b86ce/src/test-helpers.tsx#L396)?

Reply:
What I was missing was using `createBrowserHistory` to create a `History` object and then passing that into `configureStore`, but I figured that out. :-)